sklearn/model_selection/_search.py

```diff
@@ -61,6 +62,9 @@ class ParameterGrid(object):
         useful to avoid exploring parameter combinations that make no sense
         or have no effect. See the examples below.
 
+    least_significant : str or list of str, optional
```

Comment:
I don't like the name `least_significant`. But I don't think specifying the entire "search order" is appropriate. That means that the caller needs to parse the parameter grid, which makes no sense in terms of modularity.